packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -134,8 +134,10 @@ Future<Null> pub(List<String> arguments, {
   MessageFilter filter,
   String failureMessage: 'pub failed',
   @required bool retry,
-  bool showTraceForErrors: true,
+  bool showTraceForErrors,
 }) async {
+  showTraceForErrors ??= isRunningOnBot;
+
```

Comment:
we should also update the docs

Reply:
Done, PTAL

Reply:
Those docs aren't quite right (e.g. if isRunningOnBot is true but showTraceForErrors is false, we won't trace). Sorry to nitpick...

Reply:
Attempted to make them more correct